### PR TITLE
Describe how RIDs are added to a media section

### LINKED ISSUE #1896: Order of RTCRtpSendParameters.encodings is not described
In the RTCRTPSendParameters.encodings, the description is:

"A sequence containing parameters for RTP encodings of media"

In "getParameters", it says "encodings is populated based on the RIDs present in the current remote description."

Example 15 says:

pc.addTransceiver(stream.getVideoTracks()[0], {
      direction: 'sendonly',
      sendEncodings: [
        {rid: 'f'},
        {rid: 'h', scaleResolutionDownBy: 2.0},
        {rid: 'q', scaleResolutionDownBy: 4.0}
      ]

This seems to indicate that the spec-writers were thinking in terms of "highest resolution first, lowest resolution last". But this is never written down as a rule anywhere. Neither is the idea that order stays consistent between calls.

The issue came up because one lower level implementation had a built-in assumption that lowest resolution would go first and highest resolution would go last - the opposite of this example.

Possible resoutions:
a - Declare that order is not fixed; the implementation can reorder encodings whenever it wants
b - Declare that order is fixed, but resolutions can occur in any order
c - Declare that order is fixed, and that scaleResolutionDownBy has to be greater or equal to previous encoding (ascending scaleResolutionDownBy) (or the other way round)

a would seem to be the alternative that requires minimal changes to implementations.
tagging @henbos @orphis @taylor-b

### PR DESCRIPTION
**Pull request by alvestrand**
Fixes #1896

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/pull/2071.html" title="Last updated on Jan 31, 2019, 12:10 PM UTC (57a88d0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2071/bfe4b29...57a88d0.html" title="Last updated on Jan 31, 2019, 12:10 PM UTC (57a88d0)">Diff</a>

### CODE REVIEW

**Comment by alvestrand** [comment]
WG interim: Need to specify that the order is in the a=simulcast attribute (which is what has semantics in the a=simulcast spec). Order of a=rid doesn't matter.

**Comment by henbos** [comment]
@alvestrand to update it to say its about the a=simulcast:send a;b;c line instead

File: webrtc.html [hunk 1]
```diff
@@ -2571,6 +2571,18 @@ <h2>Methods</h2>
...
+                      <li>
+                        <p>If the length of the <a>[[\SendEncodings]]</a> slot
+                          of the RTCRtpSender is
+                          larger than 1, then
+                          for each encoding given in <a>[[\SendEncodings]]</a>
+                          of the RTCRtpSender, add an "a=rid send" line to the
+                          corresponding media section, in the same order as
+                          given in the "encodings" field. No RID restrictions
+                          are set.
+                        </p>
```
[the remark this answers is not in the document]
Comment (alvestrand) [reply webrtc.html]: That's a task for HTML Tidy.

File: webrtc.html [hunk 2]
```diff
@@ -2571,6 +2571,18 @@ <h2>Methods</h2>
...
                         preferences.</p>
                       </li>
+                      <li>
+                        <p>If the length of the <a>[[\SendEncodings]]</a> slot
+                          of the RTCRtpSender is
+                          larger than 1, then
+                          for each encoding given in <a>[[\SendEncodings]]</a>
+                          of the RTCRtpSender, add an "a=rid send" line to the
+                          corresponding media section, in the same order as
+                          given in the "encodings" field. No RID restrictions
```
[the remark this answers is not in the document]
Comment (alvestrand) [reply webrtc.html]: Done (I think).

**Comment by alvestrand** [comment]
Note - I have a reference to draft-ietf-mmusic-sdp-simulcast as [SDP-SIMULCAST], so this shouldn't land until we have landed https://github.com/tobie/specref/pull/503 - however, that's not what Travis is complaining about. @aboba may have to look at that.

**Pull request merged**